v1/remote/image_test.go

```diff
@@ -0,0 +1,82 @@
+package remote
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-containerregistry/authn"
+	"github.com/google/go-containerregistry/name"
+)
+
+type mockRegistry struct {
+	requests []*http.Request
+}
+
+// TODO(jonjohnsonjr): Replace this with better test data.
+const (
+	goodManifest = `{
+  "schemaVersion": 2,
+  "config": {
+    "digest": "sha256:deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef"
+  }
+}`
+
+	goodConfig = `{
+  "architecture": "amd64"
+}`
+)
+
+func TestManifestAndConfig(t *testing.T) {
+	reg := new(mockRegistry)
+	server := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			reg.requests = append(reg.requests, r)
+
+			if r.URL.Path == "/v2/" {
+				w.WriteHeader(http.StatusOK)
+				return
+			}
+
+			if strings.Contains(r.URL.Path, "manifests") {
+				w.Write([]byte(goodManifest))
+			}
+
+			if strings.Contains(r.URL.Path, "blobs") {
+				w.Write([]byte(goodConfig))
+			}
+		}))
+	defer server.Close()
+
+	inner := &http.Transport{
+		Proxy: func(req *http.Request) (*url.URL, error) {
+			return url.Parse(server.URL)
+		},
+	}
+
+	ref, err := name.NewTag("localhost:5000/library/ubuntu:latest", name.StrictValidation)
```

Comment:
If you just made `localhost:5000` => `server.URL` I think you can avoid the proxy bit.